hail/python/hailtop/batch/contrib/regenie/regenie-batch.py

```diff
@@ -0,0 +1,293 @@
+    parser.add_argument('--bsize', required=False)
+    parser.add_argument('--cv', required=False)
+    parser.add_argument('--nb', required=False)
+    parser.add_argument('--out', required=False)
+
+    parser.add_argument('--loocv', required=False, action='store_true')
+    parser.add_argument('--force-impute', required=False, action='store_true')
+    parser.add_argument('--bt', required=False, action='store_true')
+    parser.add_argument('--1', '--cc12', required=False, action='store_true')
+    parser.add_argument('--split', required=False, action='store_true')
+    parser.add_argument('--strict', required=False, action='store_true')
+    parser.add_argument('--ignore-pred', required=False, action='store_true')
+    parser.add_argument('--firth', required=False, action='store_true')
+    parser.add_argument('--approx', required=False, action='store_true')
+    parser.add_argument('--spa', required=False, action='store_true')
+    parser.add_argument('--debug', required=False, action='store_true')
+    parser.add_argument('--verbose', required=False, action='store_true')
+    parser.add_argument('--lowmem', required=False, action='store_true')
+
+    parser.add_argument('--lowmem-prefix', required=False)
+    parser.add_argument('--threads', required=False, default=1)
+
+    if not hexists(path_or_str):
+        print(f"Couldn't find a file named {path_or_str}, assuming this is an argument string")
+        r = parser.parse_args(shlex.split(path_or_str))
+
+    else:
+        print(f"Found {path_or_str}, reading")
+
+        with hopen(path_or_str, "r") as f:
+            t = shlex.split(f.read())
+            r = parser.parse_args(t)
+
+    ninputs = sum(i is not None for i in [r.bed, r.bgen, r.pgen])
+    if ninputs == 0:
+        raise Exception("An input file is required. Please specify: --bed or --bgen or --pgen")
+
+    if ninputs > 1:
+        raise Exception(f"""More than one input file specified.
+                            Check that only one of --bed or --bgen or --pgen are specified for --step {r.step}
+                        """)
+
+    if r.step == 2:
+        if r.pred:
+            print("--pred provided for --step 2, but Batch will constrain the --pred output prefix to that of --step 1.")
+
+        if r.extract or r.exclude:
+            raise Exception("--extract and --exclude only work with --step 1")
+
+    if r.lowmem and not r.lowmem_prefix:
+        raise Exception("When --lowmem provided, --lowmem-prefix required")
+
+    return r
+
+
+# TODO: check --phenoCol, --phenoColList, --remove
+def get_phenos(step_args: argparse.Namespace):
+    with hopen(step_args.phenoFile, "r") as f:
+        return f.readline().strip().split(" ")[2:]
+
+
+def get_input(batch, step_args: argparse.Namespace):
+    add = {}
+    for name, val in vars(step_args).items():
+        if name in from_underscore:
+            name = from_underscore[name]
+
+        if name not in input_file_args or val is None:
+            continue
```

Comment:
instead of if..continue..if..continue, prefer:

```
if ...:
elif ...:
elif ...:
else:
```

Reply:
There is no need for an elif after a continue, it's like elif after return